components/inputs/input-number.js

```diff
@@ -7,46 +7,51 @@ import { ifDefined } from 'lit-html/directives/if-defined.js';
 import { LocalizeCoreElement } from '../../lang/localize-core-element.js';
 import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
 
-function formatValue(value, minFractionDigits, maxFractionDigits, showTrailingZeroes, numDecimalDigits) {
-	let adjustedMinFractionDigits = minFractionDigits;
-	if (showTrailingZeroes && numDecimalDigits > 0) {
-		adjustedMinFractionDigits = Math.max(minFractionDigits || 0, numDecimalDigits);
-		if (maxFractionDigits !== undefined) {
-			adjustedMinFractionDigits = Math.min(
-				adjustedMinFractionDigits,
-				maxFractionDigits
-			);
-		}
+function formatValue(value, options, numDecimalDigits) {
+	if (value === undefined) return '';
+	if (numDecimalDigits > 0) {
+		options.minimumFractionDigits = Math.min(
+			Math.max(options.minimumFractionDigits, numDecimalDigits),
+			options.maximumFractionDigits
+		);
 	}
-	const options = {
-		maximumFractionDigits: maxFractionDigits,
-		minimumFractionDigits: adjustedMinFractionDigits
-	};
 	return formatNumber(value, options);
 }
 
-function countDecimalDigits(value) {
+function countDecimalDigits(value, useLocaleDecimal) {
 
 	if (value === undefined || value === null) {
-		return undefined;
+		return 0;
 	}
 
-	const descriptor = getNumberDescriptor();
+	const decimalSymbol = useLocaleDecimal ? getNumberDescriptor().symbols.decimal : '.';
 
 	let numDecimalDigits = 0;
 	for (let i = value.length - 1; i > -1; i--) {
 		const c = value.charAt(i);
-		if (c === descriptor.symbols.decimal) {
+		if (c === decimalSymbol) {
 			return numDecimalDigits;
 		} else {
 			numDecimalDigits++;
 		}
 	}
 
-	return undefined;
+	return 0;
 
 }
 
+function roundPrecisely(val, maxFractionDigits) {
+	const strValue = new Intl.NumberFormat(
+		'en-US',
+		{
+			maximumFractionDigits: maxFractionDigits,
+			minimumFractionDigits: 0,
+			useGrouping: false
+		}
+	).format(val);
+	return parseFloat(strValue);
```

Comment:
I had to change the rounding algorithm as the other one was failing on large numbers with the huge amount of decimal precision that the monolith requires (16 digits). A number larger than `300,000` plus 16 digits of precision was overflowing the Number boundaries.

This is actually better/simpler anyway... just use the browser's formatter and have it do the rounding, then parse that!